etc/shell.nix

```diff
@@ -1,7 +1,9 @@
 # This provides a shell with all the necesarry packages required to run mach and build servo
 # NOTE: This does not work offline or for nix-build
 
-with import <nixpkgs> {
+with import (builtins.fetchTarball {
+  url = "https://github.com/NixOS/nixpkgs/archive/6adf48f53d819a7b6e15672817fa1e78e5f4e84f.tar.gz";
```

Comment:
Are we pinning the nixpkgs for all the tools to make this independent of the user's channel state (similar to flakes)? 

The `6adf48f53d819a7b6e15672817fa1e78e5f4e84f` version is quite old and I think now that Servo has moved to stable rust, I suspect pinning GNU Make is no longer necessary [#375](https://github.com/servo/mozjs/issues/375) so we should be able to move to a newer nixpkg snapshot if needed.

Reply:
yeah, that was the idea :) i would even be open to adopting flakes later, but i’m not yet sure how that will affect the UX, since flakes are still “experimental”.

wdyt about landing with this version, then moving to a newer snapshot in a followup? that way people can use this while i test for the make and llvm regressions.

Reply:
> wdyt about landing with this version, then moving to a newer snapshot in a followup? 

yep, this sounds good to me!